### PR TITLE
Cookbook for session timers

### PR DESCRIPTION
**Pull request by martinmitrevski**
### 🔗 Issue Links

Resolves https://github.com/GetStream/ios-issues-tracking/issues/749.

### 🎯 Goal

Show an example on how to use session timers.

### 📝 Summary

_Provide bullet points with the most important changes in the codebase._

### 🛠 Implementation

_Provide a detailed description of the implementation and explain your decisions if you find them relevant._

### 🎨 Showcase

_Add relevant screenshots and/or videos/gifs to easily see what this PR changes, if applicable._

|  Before  |  After  |
| -------- | ------- |
|  `img`   |  `img`  |

### 🧪 Manual Testing Notes

For now available on ProntoStaging (available from the debug menu). Set a call duration of 2 minutes (also in the menu), and wait for a minute.

### ☑️ Contributor Checklist

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] This change follows zero ⚠️ policy (required)
- [x] This change should receive manual QA
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (Docusaurus, tutorial, CMS)

### 🎁 Meme

_Provide a funny gif or image that relates to your work on this pull request. (Optional)_

### CODE REVIEW

File: DemoApp/Sources/Components/SessionTimer/SessionTimer.swift [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    )
+    
+    let extensionTime: TimeInterval
+    
+    var showExtendCallDurationButton: Bool {
+        call?.state.ownCapabilities.contains(.changeMaxDuration) == true
+    }
+
+    @MainActor init(
+        call: Call?,
```
Comment (laevandus) [inline DemoApp/Sources/Components/SessionTimer/SessionTimer.swift:69]: If `call` is nil, is there a point to init this object at all? Seems like the call object is required.

File: docusaurus/docs/iOS/05-ui-cookbook/21-session-timers.mdx [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+
+This code will create a call, which will have a duration of 300 seconds (5 minutes), as soon as the session is started (a participant joined the call). 
+
+You can check the start date of a call with the following code:
+
+```swift
+let startedAt = call.state.session?.startedAt
+```
+
+When the `maxDuration` of a call is specified, the call session also provides the `timerEndsAt` value, which provides the date when the call will end. When a call is ended, all the participants are kicked out of the call.
```
Comment (laevandus) [inline docusaurus/docs/iOS/05-ui-cookbook/21-session-timers.mdx:27]: For me it felt like the "kicked out" is a bit too strong term:
"When a call is ended, all the participants are kicked out of the call." 

Maybe something softer like:
"When the call has ended, all the participants are removed from the call."

File: docusaurus/docs/iOS/05-ui-cookbook/21-session-timers.mdx [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+```swift
+let timerEndsAt = call.state.session?.timerEndsAt
+```
+
+## Extending the call duration
+
+You can also extend the duration of a call, both before or during the call. To do that, you should use the `call.update` method:
+
+```swift
+let newDuration = (call.state.settings?.limits.maxDurationSeconds ?? 0) + Int(extendDuration)
```
Comment (laevandus) [inline docusaurus/docs/iOS/05-ui-cookbook/21-session-timers.mdx:38]: Note: Apple uses `XYZInSeconds` quite a bit but that is already existing API so ignore the comment

**Review by laevandus — APPROVED** [review]
Some minor comments but nothing major to add.

File: DemoApp/Sources/Components/SessionTimer/SessionTimer.swift [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        rawValue: OwnCapability.changeMaxDuration.rawValue
+    )
+    
+    let extensionTime: TimeInterval
+    
+    var showExtendCallDurationButton: Bool {
+        call?.state.ownCapabilities.contains(.changeMaxDuration) == true
+    }
+
+    @MainActor init(
```
[the remark this answers is not in the document]
Comment (martinmitrevski) [reply DemoApp/Sources/Components/SessionTimer/SessionTimer.swift:68]: leftover, thx, removed it

File: DemoApp/Sources/Components/SessionTimer/SessionTimer.swift [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+                }
+            )
+        }
+    }
+    
+    deinit {
+        timer?.invalidate()
+        sessionEndCountdown?.invalidate()
+        timer = nil
+        sessionEndCountdown = nil
```
[the remark this answers is not in the document]
Comment (martinmitrevski) [reply DemoApp/Sources/Components/SessionTimer/SessionTimer.swift:139]: done

File: DemoApp/Sources/Components/SessionTimer/SessionTimer.swift (hunk 1 above)
Comment (martinmitrevski) [reply DemoApp/Sources/Components/SessionTimer/SessionTimer.swift:69]: it's optional from the view model, so best to keep it like this.

File: docusaurus/docs/iOS/05-ui-cookbook/21-session-timers.mdx (hunk 2 above)
Comment (martinmitrevski) [reply docusaurus/docs/iOS/05-ui-cookbook/21-session-timers.mdx]: fair point

File: docusaurus/docs/iOS/05-ui-cookbook/21-session-timers.mdx (hunk 3 above)
Comment (martinmitrevski) [reply docusaurus/docs/iOS/05-ui-cookbook/21-session-timers.mdx:38]: this is from the generated open api code, so we shouldn't change it.

**Pull request merged**